### PR TITLE
1.1.0 Add cancel function

### PR DESCRIPTION
**Pull request by ealush**
Allow cancelling already initiated timeouts. Useful for cases when you do not expect the debounced actions anymore.

Use it like this:

```js
const debouncer = linear({...});
debouncer();
debouncer.cancel();
```

image

### CODE REVIEW

File: playground/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const linear = require('./node_modules/linear-debounce/dist/linear');
```
Comment (omrilotan) [inline playground/index.js]: Either create this script in the npm "prepare" script:
```sh
mkdir -p playground/node_modules
ln -s ../../playground/node_modules/linear-debounce
```

Or add line:
```
!playground/node_modules
```
to you your `.gitignore` file

Comment (ealush) [reply playground/index.js]: Good call.

File: src/index.js [hunk 2]
```diff
@@ -14,7 +14,15 @@ function linear(wait: Wait): Function {
     const timeouts: Timeouts = {},
         delays: Array<string> = Object.keys(wait);
 
-    return function(...args): void {
+    function cancel(): void {
+        for (const timeout: string in timeouts) {
+            if (timeouts.hasOwnProperty(timeout)) {
```
Comment (omrilotan) [inline src/index.js]: Use of `Object.values` can save you checking `hasOwnProperty`:
```
Object.values(timeouts).forEach(clearTimeout)
```

Comment (ealush) [reply src/index.js]: Nice one. Added.

**Pull request merged**